### PR TITLE
Delete grammar comments in `CodeGeneration/SyntaxSupport` and translated all other comments to documentation

### PR DESCRIPTION
**Pull request by ahoppen**
The grammar comments were partially out-of-date and weren’t very useful and all other comments should be documentation so that clients of swift-syntax can read them.

### CODE REVIEW

**Comment by ahoppen** [comment]
@swift-ci Please test

**Review by kimdv — APPROVED** [review]

File: Sources/SwiftSyntax/generated/syntaxNodes/SyntaxNodesOP.swift [hunk 1]
```diff
@@ -1257,6 +1257,8 @@ public struct OriginallyDefinedInAttributeArgumentsSyntax: SyntaxProtocol, Synta
 
 // MARK: - PackElementExprSyntax
 
+/// A pack element expr spelled with `each`.
```
Comment (bnbarham) [inline Sources/SwiftSyntax/generated/syntaxNodes/SyntaxNodesOP.swift:1260]: `expression` instead of `expr`? Same thing with `repeat`

**Review by bnbarham — APPROVED** [review]

**Comment by ahoppen** [comment]
@swift-ci Please test

**Comment by ahoppen** [comment]
@swift-ci Please test Windows

**Pull request merged**